dspace-api/src/main/java/org/dspace/content/authority/DSpaceControlledVocabulary.java

```diff
@@ -240,4 +230,96 @@ public String getLabel(String field, String key, String locale)
     		return("");
     	}
     }
+    
+    @Override
+    public boolean isHierarchical() {
+		return true;
+	}
+    
+	@Override
+	public Choice getChoice(String fieldKey, String authKey, String locale) {
+		init();
+		log.debug("Getting matches for '" + authKey + "'");
+		String xpathExpression = String.format(idTemplate, authKey);
+		XPath xpath = XPathFactory.newInstance().newXPath();
+		try {
+			Node node = (Node) xpath.evaluate(xpathExpression, vocabulary, XPathConstants.NODE);
+			if (node != null) {
+				String[] authorities = new String[1];
+				String[] values = new String[1];
+				String[] labels = new String[1];
+				String[] parent = new String[1];
+				String[] note = new String[1];
+				readNode(authorities, values, labels, parent, note, 0, node);
+
+				if (values.length > 0) {
+					Choice choice = new Choice(authorities[0], values[0], labels[0]);
+					if (StringUtils.isNotBlank(parent[0])) {
+						choice.extras.put("parent", parent[0]);
+					}
+                    if(StringUtils.isNotBlank(note[0])) {
+                    	choice.extras.put("note", note[0]);
+                    }
+					return choice;
+				}
+			}
+		} catch (XPathExpressionException e) {
+			log.warn(e.getMessage(), e);
+		}
+		return new Choice("", "", "");
+	}
+
+	
+	private void readNode(String[] authorities, String[] values, String[] labels, String[] parent, String[] notes, int i, Node node) {
+		String hierarchy = this.buildString(node);
+		if (this.suggestHierarchy)
+		{
+		    labels[i] = hierarchy;
+		}
+		else
+		{
+		    labels[i] = node.getAttributes().getNamedItem("label").getNodeValue();
+		}
+		if (this.storeHierarchy)
+		{
+		    values[i] = hierarchy;
+		}
+		else
+		{
+		    values[i] = node.getAttributes().getNamedItem("label").getNodeValue();
+		}
+		
+		NodeList childNodes = node.getChildNodes();
+        for (int ci = 0; ci < childNodes.getLength(); ci++)
+        {
+            Node firstChild = childNodes.item(ci);
+            if(firstChild!=null && "hasNote".equals(firstChild.getNodeName())) {
+            	String nodeValue = firstChild.getTextContent();
+            	if(StringUtils.isNotBlank(nodeValue)) {
+            		notes[i] = nodeValue;
+            	}
+            }
+		}
```

Comment:
The indentation makes this hard to read.

Reply:
Noted, thank you.